pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -440,6 +446,7 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 			validate: func(c client.Client, t *testing.T) {
 				deprovision := getDeprovision(c)
 				assert.NotNil(t, deprovision, "expected deprovision request")
+				assert.Equal(t, testClusterDeployment().Name, deprovision.Labels[constants.ClusterDeploymentNameLabel], "incorrect cluster deployment name label")
```

Comment:
This should only be run when the previous assert passes.
```
if assert.NotNil(t, deprovision, ...) {
  assert.Equal(t, testClusterDeployment().Name, ...)
}
```

Reply:
May want to use testify 'require' module here, all the same assertions but the test fails immediately when one fails. (provided you don't want to keep validating other things, that may not be the case in a validate() function)

Reply:
done